### PR TITLE
[PyTorch] Add Support for aten.upsample_bilinear2d.vec

### PR DESCRIPTION
**Pull request by Atrisan**
## Description
Adds an upsample tensor node + expansion.
Fixes an implicit transpose of graph outputs in the convolution as an intermediate fix.

## Type of Change
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactor / internal change (no functional change)
- [ ] Documentation update

## Quality Checklist
*Please ensure the following are completed before requesting review:*

* [x] Runtime-compatible (no externally visible existing method has its parameters changed, was deleted or has its exception behavior changed)
* [x] Compile-time compatible (new arguments & properties added have default values, but users need to recompile against the new version to work)

IR:
* [ ] Introduces new types into the SDFG IR
* [x]  Changed / added properties on SDFG IR Elements (LibraryNodes, ControlFlowNodes DataFlowNodes etc.)
  *  [x] New / modified types have serialization / deserialization support
  *  [x] Deserialization is backward compatible (new serialization can deserialize old format and represent it in the new format if applicable)
  *  [x] Serialized format is backward compatible (previous deserializer can read new serialized format without incorrectness (just lacking additional guarantees that are optional or did not exist before)
  *  [ ] Serializer has an option to emit the previous format

### Unit Tests
- [x] New unit tests have been added for the changes.
- [x] Existing unit tests pass locally.
- [x] Edge cases and error paths are covered.

### Integration Tests
- [x] New integration tests have been added (or existing ones updated).
- [x] Integration tests pass locally.
- [x] Interactions with other components/services are verified.

### Documentation
- [ ] Public APIs, options, and behavior changes are documented.
- [x] README / docs site / inline comments updated where applicable.
- [ ] Changelog / release notes updated (if applicable).

### Test Coverage
- [ ] Test coverage has not decreased as a result of this change.
- [ ] Coverage report reviewed and acceptable.
- [ ] Critical / new code paths are covered.

## How Has This Been Tested?
Describe the tests you ran to verify your changes and provide instructions to reproduce.
Include relevant details for your test configuration (OS, Python/LLVM/Docc version, hardware/target).

## Additional Context
Add any other context about the PR here (design decisions, trade-offs, follow-ups).

### CODE REVIEW

File: pytorch/docc/pytorch/graph_parser/vision.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+            )
+        align_corners: bool = align_corners_arg
+        scale_factors: list[float] = []
+        if len(node.args) == 3 and node.args[2] is None:
+            raise GraphParserError(
+                self,
+                node,
+                "Scale factors cannot be None if sizes are not specified: "
+                + str(node.args[2]),
+            )
```
Comment (WashimNeupane) [inline pytorch/docc/pytorch/graph_parser/vision.py:65]: wouln't arg[2] still be align_corners here even when 3 arguments are provided? it would never enter this error in that case

**Review by WashimNeupane — APPROVED** [review]
